src/rgw/rgw_keystone.h

```diff
@@ -95,28 +95,33 @@ class KeystoneToken {
   void decode_json(JSONObj *access_obj);
 };
 
+
 class RGWKeystoneTokenCache {
   struct token_entry {
     KeystoneToken token;
     list<string>::iterator lru_iter;
   };
 
-  CephContext *cct;
+  CephContext * const cct;
 
   string admin_token_id;
   map<string, token_entry> tokens;
   list<string> tokens_lru;
 
   Mutex lock;
 
-  size_t max;
+  const size_t max;
 
-public:
-  RGWKeystoneTokenCache(CephContext *_cct, int _max)
-    : cct(_cct),
+  RGWKeystoneTokenCache()
+    : cct(g_ceph_context),
       lock("RGWKeystoneTokenCache", true /* recursive */),
-      max(_max) {
+      max(cct->_conf->rgw_keystone_token_cache_size) {
   }
+public:
+  RGWKeystoneTokenCache(const RGWKeystoneTokenCache&) = delete;
+  void operator=(const RGWKeystoneTokenCache&) = delete;
+
+  static RGWKeystoneTokenCache& get_instance();
 
   bool find(const string& token_id, KeystoneToken& token);
```

Comment:
This method does more than just finding a token ;-). It removes a token from the cache if it's expired.